### PR TITLE
adding reference tests for protectionsState

### PR DESCRIPTION
**Pull request by shakyShane**
Asana: https://app.asana.com/0/0/1205839123798774/f

Added tests for the new `protectionsState: 0 | 1' param in the breakage form

- Extension branch where this was tested: 
  - https://github.com/duckduckgo/duckduckgo-privacy-extension/tree/shane/ref-tests
- Android, same
  - https://github.com/duckduckgo/Android/pull/3821
  - Passing test run https://github.com/duckduckgo/Android/actions/runs/6785317611

### CODE REVIEW

**Comment by shakyShane** [comment]
Current dependencies on/for this PR:
* main
  * **PR #112** <a href="https://app.graphite.dev/github/pr/duckduckgo/privacy-reference-tests/112" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/privacy-reference-tests/112?utm_source=stack-comment).

File: broken-site-reporting/multiple_report_tests.json [hunk 1]
```diff
@@ -23,7 +24,8 @@
                             {"name": "tds", "value": "abc123"},
                             {"name": "blockedTrackers", "value": "bad.tracker.test,tracking.test"},
                             {"name": "surrogates", "value": "surrogate.domain.test,domain2.test"},
-                            {"name": "lastSentDay", "present": false}
+                            {"name": "lastSentDay", "present": false},
+                            {"name": "protectionsState", "value": "1"}
```
Comment (kdzwinel) [inline broken-site-reporting/multiple_report_tests.json:28]: oh boy, we have `"1"`, `"true"` and `false` in the params list 😬 Can we still make `protectionsState` a proper boolean?

File: broken-site-reporting/tests.json [hunk 2]
```diff
@@ -13,17 +13,74 @@
...
+                    {"name": "upgradedHttps", "value": "true"},
+                    {"name": "tds", "value": "abc123"},
+                    {"name": "blockedTrackers", "value": "bad.tracker.test,tracking.test"},
+                    {"name": "surrogates", "value": "surrogate.domain.test,domain2.test"},
+                    {"name": "protectionsState", "value": "0"}
+                ],
+                "exceptPlatforms": []
+            },
+            {
+                "name": "Denylisted (extension only)",
```
Comment (kdzwinel) [inline broken-site-reporting/tests.json:54]:
```suggestion
                "name": "Protections force-enabled by the user",
```

Support may change in the future, I think it's enough to express it via `exceptPlatforms`.

Great idea for a test BTW, glad you are thinking about those edge-cases.

File: broken-site-reporting/README.md [hunk 3]
```diff
@@ -34,6 +34,11 @@ Test suite specific fields:
 - `remoteConfigEtag` - string - string representation of remote configuration etag
 - `remoteConfigVersion` - string - string representation of remote configuration version (note, this is the numeric version found in the remote config (e.g, `1680178584671`, not `v1` or `v2`))
 - `providedDescription` - string - user-provided breakage description
+- `protectionsEnabled` - boolean - if protections are enabled (true) or disabled (false). Reasons for being disabled could be:
+    - current domain exists in the user's allow-list
+    - current domain exists in the 'unprotectedTemporary' section of the privacy-configuration
+    - current domain is an exception for the `contentBlocking` feature in the privacy-configuration
+    - the feature `contentBlocking` is disabled via privacy-configuration
```
Comment (kdzwinel) [inline broken-site-reporting/README.md:41]:
```suggestion
- `protectionsEnabled` - boolean - if protections are enabled (true) or disabled (false) as visible to the user on the dashboard.
```
I think we can drop the reasons from here. Those are explained in the spec and it will be easier to keep them updated if we have them in one central place.

"As visible" part should make it clear that the whole "reasons for being disabled" logic applies.

**Review by kdzwinel — COMMENTED** [review]
Thanks Shane! I left couple of comments.

File: broken-site-reporting/multiple_report_tests.json (hunk 1 above)
Comment (shakyShane) [reply broken-site-reporting/multiple_report_tests.json:28]: it's already implemented across all platforms as `"1" | "0"` - but it would be trivial to change it now if we'd prefer? Probably better for me to update platforms now, as part of this project if we regret the "1" | "0".

I think I was under the impression the choice was discussed etc, I could've/should've questioned it perhaps.

**Comment by shakyShane** [comment]
@kdzwinel  I made those changes! 

yeah the extension PR is expected to fail. Once this is merged then the extension PR will be merged after to consume the correct version and we'll be all good.

I've also verified this change will be ok on Android https://github.com/duckduckgo/Android/actions/runs/6823524571
